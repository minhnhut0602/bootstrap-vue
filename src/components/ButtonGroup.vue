<template>
    <div :class="classes">
        <slot />
    </div>
</template>

<script lang="ts" setup>
import {computed, defineProps} from 'vue';
import useSize, {sizeProps} from '@/composables/useSize';
import useClasses from '@/composables/useClasses';

const props = defineProps({
    ...sizeProps,
    vertical: {
        type: Boolean,
        default: false,
    },
});

const {classes} = useClasses([
    computed(() => props.vertical ? 'btn-group-vertical' : 'btn-group').value,
    useSize(props.size, 'btn-group-{0}').sizeClass.value,
]);
</script>
