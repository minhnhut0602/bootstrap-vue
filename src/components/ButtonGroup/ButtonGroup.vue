<template>
    <div :class="classes">
        <slot />
    </div>
</template>

<script lang="ts">
import {sizeProps} from '../../composables/useSize';
</script>

<script lang="ts" setup>
import {computed} from 'vue';
import useClasses from '../../composables/useClasses';
import useSize from '../../composables/useSize';

const props = defineProps({
    ...sizeProps,
    vertical: {
        type: Boolean,
        default: false,
    },
});

const {classes} = useClasses([
    computed(() => props.vertical ? 'btn-group-vertical' : 'btn-group').value,
    useSize(props.size, 'btn-group-{0}').sizeClass.value,
]);
</script>
