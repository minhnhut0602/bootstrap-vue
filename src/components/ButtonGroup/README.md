# ButtonGroup

```html
<ButtonGroup vertical>
    <Button>
        Top
    </Button>

    <Button>
        Middle
    </Button>

    <Button>
        Bottom
    </Button>
</ButtonGroup>
```

## Props

| Name | Type | Description |
| - | - | - |
| size | [Size](#link-to-size-definition) (string) | The size of the button group. |
| verical | boolean | Stacks the buttons vertically. |

## Slots

| Name | Description |
| - | - |
| default | Displays content inside the button group. |
