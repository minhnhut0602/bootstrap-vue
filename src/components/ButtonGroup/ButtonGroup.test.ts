import ButtonGroup from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(ButtonGroup);

    componentSlotRenderTest(ButtonGroup);

    componentWrapperClassTest(ButtonGroup, {vertical: true}, 'btn-group-vertical');

    componentWrapperClassTest(ButtonGroup, {size: 'lg'}, 'btn-group-lg');
});
