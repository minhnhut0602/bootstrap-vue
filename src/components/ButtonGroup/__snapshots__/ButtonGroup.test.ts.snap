// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `"<div class=\\"btn-group btn-group-md\\"></div>"`;
